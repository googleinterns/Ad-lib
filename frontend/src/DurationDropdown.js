import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import NativeSelect from '@material-ui/core/NativeSelect';

const durations = [
  { label: '15 minutes', value: 15 },
  { label: '30 minutes', value: 30 },
  { label: '45 minutes', value: 45 },
  { label: '60 minutes', value: 60 }
];

export default function DurationDropdown(props) {
  return (
    <div>
      <FormControl style={{width: 180}}>
        <NativeSelect
          id="duration-input"
          name="duration"
          onChange={event => props.onChange(event.target.value)}
          inputProps={{ "aria-label": "duration" }}
        >
          {durations.map((currentDuration) => (
            <option value={currentDuration.value}>
              {currentDuration.label}
            </option>
          ))}  
        </NativeSelect>
      </FormControl>
    </div>
  );
}
