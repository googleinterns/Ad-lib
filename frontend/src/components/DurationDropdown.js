import React from 'react';
import PropTypes from 'prop-types';
import FormControl from '@material-ui/core/FormControl';
import NativeSelect from '@material-ui/core/NativeSelect';

const durations = [
  {label: '15 minutes', value: 15},
  {label: '30 minutes', value: 30},
  {label: '45 minutes', value: 45},
  {label: '60 minutes', value: 60},
];

// Add onChange to props validation
DurationDropdown.propTypes = {
  onChange: PropTypes.func,
};

/**
 * Create duration dropdown component using durations array
 * @return {DurationDropdown} DurationDropdown component
 * @param {Object} props
 */
export default function DurationDropdown(props) {
  return (
    <div>
      <FormControl style={{width: 180}}>
        <NativeSelect
          id="duration-input"
          name="duration"
          onChange={(event) => props.onChange(event.target.value)}
          inputProps={{'aria-label': 'duration'}}
        >
          {durations.map((currentDuration) => (
            <option key={currentDuration.label} value={currentDuration.value}>
              {currentDuration.label}
            </option>
          ))}
        </NativeSelect>
      </FormControl>
    </div>
  );
}
