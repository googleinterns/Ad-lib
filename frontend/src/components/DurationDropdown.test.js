import React from 'react';
import DurationDropdown from './DurationDropdown';
import renderer from 'react-test-renderer';

describe('Duration Dropdown', () => {
  it('should be defined', () => {
    expect(DurationDropdown).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = renderer.create(<DurationDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});  