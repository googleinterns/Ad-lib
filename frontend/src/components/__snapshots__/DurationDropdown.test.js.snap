// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duration Dropdown should render correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root"
    style={
      Object {
        "width": 180,
      }
    }
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-label="duration"
        aria-labelledby="duration-input"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="duration-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        15 minutes
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        name="duration"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value={15}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div>
  <div
    class="MuiFormControl-root"
    style="width:180px;"
  >
    <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl">
      <div
        aria-haspopup="listbox"
        aria-label="duration"
        aria-labelledby="duration-input"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="duration-input"
        role="button"
      >
        15 minutes
      </div>
      <input
        aria-hidden
        class="MuiSelect-nativeInput"
        name="duration"
      >
      <svg
        aria-hidden
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path d="M7 10l5 5 5-5z"></path>
      </svg>
    </div>
  </div>
</div>
`;
